src/pages/keyword-generator/KeywordGenerator.vue

```diff
@@ -1,9 +1,100 @@
 <script setup>
+import { ref, computed } from 'vue';
+
+const text = ref('');
+const keywords = ref([]);
+
+const generateWordArray = (keywords, n) => {
+  const result = new Set();
+  const words = keywords.split(" ");
+  for (let i = 0; i <= words.length - n; i++) {
+    const nGram = words.slice(i, i + n).join(' ');
+    result.add(nGram);
+  }
+  return Array.from(result);
+}
+
+const generateNGrams = () => {
+  keywords.value = [];
+  if (!text.value.trim()) { return; }
+  for(let i = 1;i <= 3; i++) {
+    keywords.value.push({i,keywords: generateWordArray(text.value, i)});
+  }
+}
+const validNGrams = computed(()=> {
+  return keywords.value.filter(nGram => nGram.keywords.length);
+});
```

Comment:
We can pass this computed property directly to the `generateNGrams` function. No need to filter it afterwards.